test/_init_tests.js

```diff
@@ -207,6 +207,49 @@ describe.configure = function() {
   return new TestConfig(describe);
 };
 
+function enableYield(fn, message, runnable) {
+  if (!runnable || !runnable.constructor
+      || runnable.constructor.name !== 'GeneratorFunction') {
+    return fn(message, runnable);
+  }
+  return fn(message, done => {
+    const runner = (iterator, result) => {
```

Comment:
We can refactor this so Promises take care of everything:

```js
// Adapted from Babel's asyncToGenerator
return realIt(message, () => {
  return new Promise((resolve, reject) => {
    const gen = runnable();
    function step(key, arg) {
      let info;
      try {
        info = gen[key](arg);
      } catch (error) {
        reject(error);
        return;
      }

      const { value, done } = info;
      if (done) {
        resolve(value);
      } else {
        Promise.resolve(value).then(_next, _throw);
      }
    }
    function _next(value) {
      step("next", value);
    }
    function _throw(err) {
      step("throw", err);
    }
    _next();
  });
});
```

Differences include:
- Promises handle async (through assimilation)
- Yielded promises can reject, going into the generators throw path
- We avoid using `done` https://github.com/ampproject/amphtml/issues/8966
- Less closures